src/dependency-manager/src/component-config/builder.ts

```diff
@@ -104,6 +104,12 @@ export class ComponentConfigBuilder {
   }
 
   static buildFromJSON(obj: any): ComponentConfig {
+    for (const key of Object.keys(obj)) {
```

Comment:
Both of these should go below the safeguard check `if (!(obj instanceof Object)) {`